### PR TITLE
Created a window trimmer, different than q-trim since it is looking a…

### PR DESCRIPTION
**Pull request by dstreett**
Another trimmer @joe-angell . This one is a window achieving a avg quality score greater than X rather than a single bp being greater. Not really sure which one works better, but people might want both. Let me know what you think. :) (Sorry today is going to be a trimmer day)

### CODE REVIEW

File: Q-Window-Trim/src/q_window_trim.cpp [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+                                           "Prefix for outputted files")
+            ("no-left,l", po::bool_switch(&no_left)->default_value(false),    "Turns of trimming of the left side of the read")
+            ("no-right,r", po::bool_switch(&no_right)->default_value(false),    "Turns of trimming of the right side of the read")
+            ("stranded,s", po::bool_switch(&stranded)->default_value(false),    "If R1 is orphaned, R2 is RC (for stranded RNA)")
+            ("window_size,w", po::value<size_t>(&window_size)->default_value(10),    "Min base pairs trim for AT tail")
+            ("avg_qual,q", po::value<size_t>(&avg_qual)->default_value(20),    "Max amount of mismatches allowed in trimmed area")
+            ("min-length,m", po::value<size_t>(&min_length)->default_value(50),    "Min length for acceptable outputted read")
+            ("log-file,L",                 "Output-Logfile")
+            ("no-log,N",                   "No logfile <outputs to stderr>")
+            ("help,h",                     "Prints help.");
```
Comment (joe-angell) [inline Q-Window-Trim/src/q_window_trim.cpp]: is all these program options and stuff duplicated?  Maybe we should move that into common.  Seems like all these programs have the same basic main, we should move that all to common to reduce the duplication.

File: Q-Window-Trim/src/q_window_trim.h [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    std::string qual = rb.get_qual();
+    int current_sum = 0;
+    int cut = -1;
+
+    for (int i = 0; i < qual.length(); ++i) {
+         
+        current_sum += qual[i];
+
+        if (i > window_size - 1) { //once we hit window size, subtract the first value off
```
Comment (joe-angell) [inline Q-Window-Trim/src/q_window_trim.h]: if (i >= window_size) is a bit more clear

File: Q-Window-Trim/src/q_window_trim.h [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        }
+        if (current_sum >= sum_qual) {
+            cut = (i - window_size + 1);
+            break;
+        }
+    }
+    if (cut == -1) {
+        cut = qual.length() ;
+    }
+    rb.setLCut(std::max(0, cut));
```
Comment (joe-angell) [inline Q-Window-Trim/src/q_window_trim.h]: nice use of max, ;)

File: Q-Window-Trim/src/q_window_trim.h [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            current_sum -= qual[i + window_size];
+        }
+        
+        if (current_sum >= sum_qual) {
+            cut = i + window_size - 1;
+            break;
+        }
+    }
+
+    if (cut == -1) {
```
Comment (joe-angell) [inline Q-Window-Trim/src/q_window_trim.h]: why no max here?

Comment (dstreett) [inline Q-Window-Trim/src/q_window_trim.h]: I did that check in a weird way I'm not happy with. Check it out now. I also had to change the sub_* returns.

**Comment by joe-angell** [comment]
yeah looks better, nice.

**Pull request merged**